app/csv_generator/csv_generator.rb

```diff
@@ -59,6 +64,13 @@ def csv_data(obj, is_well_header_row, sample_names)
     end
   end
 
+  def check_template_prep_kit_box_barcode(well_libraries)
```

Comment:
Might be better to put this method in well. That way you don't need any logic in the csv generator. Then you can just have template_prep_kit_box_barcode in pipelines.yml